provider/image.go

```diff
@@ -253,7 +254,10 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 
 	//Start a session for BuildKit
 	if build.BuilderVersion == defaultBuilder {
-		sess, _ := session.NewSession(ctx, "pulumi-docker", "")
+		sess, err := session.NewSession(ctx, "pulumi-docker", identity.NewID())
```

Comment:
What does passing `identity.NewID()` do? Sorry if didn't parse it from your PR description.

Reply:
The most complete permissively licensed client I've found is the Tilt tool, here:

https://github.com/tilt-dev/tilt/blob/a0891ef56f2d501969a27e1ecc781e2c8961cc0a/cmd/buildkitapi/main.go#L53

While we aren't using all of the same Buildkit options, I couldn't find any documentation on what the third parameter, `SharedKey`, does.

But it seems to me that we likely want to avoid any inadvertent sharing between concurrent Docker builds, so I followed the Tilt API's here.

I used this GitHub Code Search to try to find areas where we might deviate from norms: https://github.com/search?q=NewSession+DialHijack+lang%3Ago+-org%3Ahashicorp+-is%3Afork&type=code

Reply:
I still don't entirely follow what the NewID() does for us.

The [source code](https://github.com/moby/buildkit/blob/master/identity/randomid.go#L38) says that it generates a random identifier for cases where such are required. Is that so here, for our session?

More specifically - do we need this change for skipped builds to no longer happen?

Either way, it seems relatively low risk to add this.